flink-dist/src/main/flink-bin/mesos-bin/mesos-appmaster.sh

```diff
@@ -38,9 +38,14 @@ constructAppMasterClassPath() {
     echo $CC_CLASSPATH:$INTERNAL_HADOOP_CLASSPATHS
 }
 
+if [ "$FLINK_IDENT_STRING" = "" ]; then
+    FLINK_IDENT_STRING="$USER"
+fi
+
+
 CC_CLASSPATH=`manglePathList $(constructAppMasterClassPath)`
 
-log=flink-appmaster.log
+log="${FLINK_LOG_DIR}/flink-${FLINK_IDENT_STRING}-mesos-appmaster-${HOSTNAME}.log"
```

Comment:
Looks good. This is inline with the other scripts we have.

---

As a small note: If for some weird reason the `FLINK_LOG_DIR` is not correctly set, this will try to create the log file in root instead of the current working directory or something else. Having said that, not relevant for this PR as it's the same for all scripts.